project/src/hooks/useMap.tsx

```diff
@@ -9,28 +9,35 @@ function useMap(
   const [map, setMap] = useState<Map | null>(null);
 
   useEffect(() => {
-    if (mapRef.current !== null && map === null) {
-      const instance = new Map(mapRef.current, {
-        center: {
-          lat: city.location.latitude,
-          lng: city.location.longitude
-        },
-        zoom: city.location.zoom
-      });
-
-      const layer = new TileLayer(
-        'https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png',
-        {
-          attribution:
-            '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>'
-        }
-      );
-
-      instance.addLayer(layer);
-
-      setMap(instance);
+    if (mapRef.current === null) {
+      return;
     }
-  }, [mapRef, map, city]);
+
+    const instance = new Map(mapRef.current, {
+      center: {
+        lat: city.location.latitude,
```

Comment:
```suggestion
        lat: location.latitude,
```
Должно зависеть от `deps` эффекта